core/src/services/dbfs/backend.rs

```diff
@@ -0,0 +1,318 @@
+            let decoded_response =
+                serde_json::from_slice::<DbfsStatus>(&stat_resp.into_body().bytes().await?)
+                    .map_err(new_json_deserialize_error)?;
+            meta.set_last_modified(parse_datetime_from_from_timestamp_millis(
+                decoded_response.modification_time,
+            )?);
+            meta.set_mode(if decoded_response.is_dir {
+                EntryMode::DIR
+            } else {
+                EntryMode::FILE
+            });
+            if !decoded_response.is_dir {
+                meta.set_content_length(decoded_response.file_size as u64);
+            }
+        }
+
+        let op = DbfsReader::new(self.core.clone(), args, path.to_string());
+
+        Ok((RpRead::with_metadata(meta), op))
+    }
+
+    async fn write(&self, path: &str, args: OpWrite) -> Result<(RpWrite, Self::Writer)> {
+        Ok((
+            RpWrite::default(),
+            oio::OneShotWriter::new(DbfsWriter::new(self.core.clone(), args, path.to_string())),
+        ))
+    }
+
+    async fn rename(&self, from: &str, to: &str, _args: OpRename) -> Result<RpRename> {
+        self.core.dbfs_ensure_parent_path(to).await?;
+
+        let resp = self.core.dbfs_rename(from, to).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK => {
+                resp.into_body().consume().await?;
+                Ok(RpRename::default())
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn stat(&self, path: &str, _: OpStat) -> Result<RpStat> {
+        // Stat root always returns a DIR.
+        if path == "/" {
+            return Ok(RpStat::new(Metadata::new(EntryMode::DIR)));
+        }
+
+        let resp = self.core.dbfs_get_status(path).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK => {
+                let mut meta = parse_into_metadata(path, resp.headers())?;
+                let bs = resp.into_body().bytes().await?;
+                let decoded_response = serde_json::from_slice::<DbfsStatus>(&bs)
+                    .map_err(new_json_deserialize_error)?;
+                meta.set_last_modified(parse_datetime_from_from_timestamp_millis(
+                    decoded_response.modification_time,
+                )?);
+                match decoded_response.is_dir {
+                    true => meta.set_mode(EntryMode::DIR),
+                    false => {
+                        meta.set_mode(EntryMode::FILE);
+                        meta.set_content_length(decoded_response.file_size as u64)
+                    }
+                };
+                Ok(RpStat::new(meta))
+            }
+            StatusCode::NOT_FOUND if path.ends_with('/') => {
+                Ok(RpStat::new(Metadata::new(EntryMode::DIR)))
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    /// NOTE: Server will return 200 even if the path doesn't exist.
```

Comment:
This NOTE seems not align with the behavior that `ErrorKind::NotFound => Ok(RpDelete::default()),`?

Reply:
Right! dropped `NOTE: Server will return 200 even if the path doesn't exist.`

Reply:
Dropped it and use only `_ => Err(parse_error(resp).await?)`.
(Yes, but I'm not sure why the [document](https://docs.databricks.com/api/azure/workspace/dbfs/delete) still lists a 404 in the possible response status. I've tried various combinations, but they all return a 200.)